File: C++/Problems/Problems/problem_769.cpp
#include "problem_769.h"

bool problem_769::test() {
    std::vector arr { 1, 0, 2, 3, 4 };

    int expected = 4;

    auto result = maxChunksToSorted(arr);

    std::cout << result << std::endl;

    return result == expected;
}

int problem_769::maxChunksToSorted(std::vector<int> &arr) {
    int n = arr.size();
    int m = -1;
    int result = 0;
    for (int i = 0; i < n; ++i)
    {
        m = std::max(arr[i], m);
        if (m == i)
            result++;
    }

    return result;
}

File: Python/Problems/Problems/problem769.py
import heapq
import math
import sys

import numpy as np

from utils import print1DArray, print2DArray
from typing import List, Set, Optional
from collections import Counter

from .problem import Problem

class Problem769(Problem):
    def test(self) -> bool:
        arr = [1, 0, 2, 3, 4]

        expected = 4

        result = self.maxChunksToSorted(arr)

        print(result)

        return result == expected

    def maxChunksToSorted(self, arr: List[int]) -> int:
        n = len(arr)
        m = -1
        result = 0
        for i in range(n):
            m = max(arr[i], m)
            if m == i:
                result += 1

        return result

